internal/heartbeat/heartbeat.go

```diff
@@ -124,12 +139,16 @@ func (s *HeartbeatData) getMutableHardwareInfoDelta(currentMutableHwInfo models.
 }
 
 type Heartbeat struct {
-	ticker           *time.Ticker
-	dispatcherClient pb.DispatcherClient
-	data             *HeartbeatData
-	lock             sync.Mutex
-	reg              *registration.Registration
-	firstHearbeat    bool
+	ticker                *time.Ticker
+	dispatcherClient      pb.DispatcherClient
+	data                  *HeartbeatData
+	lock                  sync.Mutex
```

Comment:
my 2 cents. 
You could compose your struct with `sync.Mutex` like
```
type Heartbeat struct {
   sync.Mutext
}
```
and lock it with `s.Lock()`. IMO it is cleaner than `s.lock.Lock()`.

Reply:
BTW: I'm a bit against this one, having a lock like this means that someone from an external package can lock the struct and deadlocks can happen without notice.

I'm ok to have it, but I remember to debug a hard deadlock because of a lock from another package.

Reply:
yeah mutex shall be private, it will be a nightmare to debug in case of issue otherwise